### PR TITLE
ros tab to use fed modules

### PR DESCRIPTION
**Pull request by kgaikwad**
ROS side PR - https://github.com/RedHatInsights/ros-frontend/pull/40

+ adding @karelhala into loop for review.

### CODE REVIEW

File: src/store/reducers.js [hunk 1]
```diff
@@ -52,7 +53,13 @@ function entityLoaded(state, { payload: { entitlements } } = { payload: {} }) {
                 title: 'Patch',
                 name: 'patch',
                 component: PatchTab
+            },
+            isEntitled(entitlements && entitlements.insights) && {
```
Comment (karelhala) [inline src/store/reducers.js:57]: Is the `ROS` enabled for all users? If not we might want to add the cookie check as well here.

File: src/components/inventory/Ros.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    const { params } = useRouteMatch('/:inventoryId');
+    const { getRegistry } = useContext(RegistryContext);
+
+    return <AsyncComponent
+        appName="ros"
+        module="./SystemDetail"
+        getRegistry={getRegistry}
+        fallback={fallback}
+        customItnlProvider
+        entity={ { id: params.inventoryId } }
```
Comment (karelhala) [inline src/components/inventory/Ros.js:17]: If you are going to use the solution with `useSelector` you can change the `entity` prop to `inventoryId={ params.inventoryId }`

Also no need for `customItnlProvider` as ROS is not using currently any intl service, right?

File: src/store/reducers.js (hunk 1 above)
Comment (kgaikwad) [reply src/store/reducers.js]: Ros is not enabled for all users. Added cookie check.

File: src/components/inventory/Ros.js (hunk 2 above)
Comment (kgaikwad) [reply src/components/inventory/Ros.js]: Done!

File: src/store/reducers.js [hunk 3]
```diff
@@ -32,6 +33,10 @@ function entitiesLoaded(state, { payload }) {
 }
 
 function entityLoaded(state, { payload: { entitlements } } = { payload: {} }) {
+
+    const hasRosCookie = insights.chrome.visibilityFunctions.isProd() ?
+        insights.chrome.visibilityFunctions.hasCookie('cs_ros_beta_enable') : true;
```
Comment (karelhala) [inline src/store/reducers.js:38]: This one should also pass in the value
```suggestion
        insights.chrome.visibilityFunctions.hasCookie('cs_ros_beta_enable', '1') : true;
```

Comment (kgaikwad) [reply src/store/reducers.js]: Thank you! Good catch :+1:

**Review by karelhala — APPROVED** [review]

File: src/store/reducers.js (hunk 1 above)
Comment (kgaikwad) [reply src/store/reducers.js]: @karelhala,
I forgot to mention one thing earlier that ROS is not enabled by default like few other services and designed for RHEL instances from cloud (AWS/azure). 
Along with cookie check, do I need to add extra conditions here? Or it is ok to show Tab with empty recommendations table.

**Review by karelhala — APPROVED** [review]

**Pull request merged**